lib/dry/system/container.rb

```diff
@@ -249,7 +249,7 @@ def finalize!(&block)
           manual_registrar.finalize!
           auto_registrar.finalize!
 
-          freeze
+          self.freeze if freeze
```

Comment:
I think the first line of the method takes that into consideration `return self if frozen?`